src/websocket/handlers/logout.cpp

```diff
@@ -32,30 +34,35 @@ void logout(
         return tr.send_error("Internal error.");
     }
 
-    if(username && username.value() == ctx.username)
+    const auto credentials_correct
+            = std::visit(
+                    meta::overloaded{
+                            [&username, &req](const request_context::user_info& cln_ctx)
+                            {
+                                return username
+                                       && username.value() == cln_ctx.username
+                                       && req.token == cln_ctx.active_token;
+                            },
+                            [](const request_context::guest_info& cln_ctx)
+                            {
+                                return false;
+                            }},
+                    ctx.cln_info);
+
+    if(!credentials_correct)
     {
-        if(req.token == ctx.active_token)
```

Comment:
You should clear ```ctx.active_token``` and ```ctx.username``` because they persists for all requests in same websocket connection.